tests/Operation/WatchFunctionalTest.php

```diff
@@ -0,0 +1,340 @@
+        $changeStream = $operation->execute($this->getPrimaryServer());
+
+        $this->assertFalse($changeStream->valid());
+        $this->assertNull($changeStream->key());
+
+        $this->insertDocument(['_id' => 1, 'x' => 'foo']);
+
+        $changeStream->rewind();
+        $this->assertTrue($changeStream->valid());
+        $this->assertSame(0, $changeStream->key());
+
+        $changeStream->next();
+        $this->assertFalse($changeStream->valid());
+        $this->assertNull($changeStream->key());
+
+        $changeStream->next();
+        $this->assertFalse($changeStream->valid());
+        $this->assertNull($changeStream->key());
+
+        $this->killChangeStreamCursor($changeStream);
+
+        $changeStream->next();
+        $this->assertFalse($changeStream->valid());
+        $this->assertNull($changeStream->key());
+
+        $this->insertDocument(['_id' => 2, 'x' => 'bar']);
+
+        $changeStream->next();
+        $this->assertTrue($changeStream->valid());
+        $this->assertSame(1, $changeStream->key());
+    }
+
+    public function testNonEmptyPipeline()
+    {
+        $pipeline = [['$project' => ['foo' => [0]]]];
+
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), $pipeline, ['maxAwaitTimeMS' => 100]);
+        $changeStream = $operation->execute($this->getPrimaryServer());
+
+        $this->insertDocument(['_id' => 1]);
+
+        $changeStream->rewind();
+        $this->assertTrue($changeStream->valid());
+
+        $expectedResult = [
+            '_id' => $changeStream->current()->_id,
+            'foo' => [0],
+        ];
+
+        $this->assertSameDocument($expectedResult, $changeStream->current());
+    }
+
+    public function testInitialCursorIsNotClosed()
+    {
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), []);
+        $changeStream = $operation->execute($this->getPrimaryServer());
+
+        /* The spec requests that we assert that the cursor returned from the
+         * aggregate command is not closed on the driver side. We will verify
+         * this by checking that the cursor ID is non-zero and that libmongoc
+         * reports the cursor as alive. While the cursor ID is easily accessed
+         * through ChangeStream, we'll need to use reflection to access the
+         * internal Cursor and call isDead(). */
+        $this->assertNotEquals('0', (string) $changeStream->getCursorId());
+
+        $rc = new ReflectionClass('MongoDB\ChangeStream');
+        $rp = $rc->getProperty('csIt');
+        $rp->setAccessible(true);
+
+        $iterator = $rp->getValue($changeStream);
+
+        $this->assertInstanceOf('IteratorIterator', $iterator);
+
+        $cursor = $iterator->getInnerIterator();
+
+        $this->assertInstanceOf('MongoDB\Driver\Cursor', $cursor);
+        $this->assertFalse($cursor->isDead());
```

Comment:
I asked @ajdavis in Slack about whether the alive boolean being `true` and the cursor ID being non-zero are equivalent in libmongoc for an already-executed cursor. For the time being, I wanted to check both.